src/Umbraco.Cms.Integrations.SEO.Semrush/Api/Management/Controllers/Token/ValidateTokenController.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Asp.Versioning;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using System.Net;
+using Umbraco.Cms.Integrations.SEO.Semrush.Configuration;
+using Umbraco.Cms.Integrations.SEO.Semrush.Models.Dtos;
+using Umbraco.Cms.Integrations.SEO.Semrush.Services;
+
+namespace Umbraco.Cms.Integrations.SEO.Semrush.Api.Management.Controllers.Token
+{
+    public class ValidateTokenController : TokenControllerBase
+    {
+        public ValidateTokenController(IOptions<SemrushSettings> options, IWebHostEnvironment webHostEnvironment, ISemrushTokenService semrushTokenService, ICacheHelper cacheHelper, TokenBuilder tokenBuilder, SemrushComposer.AuthorizationImplementationFactory authorizationImplementationFactory) : base(options, webHostEnvironment, semrushTokenService, cacheHelper, tokenBuilder, authorizationImplementationFactory)
+        {
+        }
+
+        [HttpGet("validate")]
+        [ProducesResponseType(typeof(AuthorizationResponseDto), StatusCodes.Status200OK)]
+        public async Task<IActionResult> ValidateToken()
+        {
+            _semrushTokenService.TryGetParameters(Constants.TokenDbKey, out TokenDto token);
+
+            if (!token.IsAccessTokenAvailable) return Ok(new AuthorizationResponseDto { IsAuthorized = false });
```

Comment:
With the new API endpoints pattern, I think it would make sense and be logical to return an `Unauthorized` response.

Reply:
I decided to not return `Unauthorized`, because it will force user to login again, and again because it keeps returning `Unauthorized` if `token.IsAccessTokenAvailable` is `false`